docs/graph-homomorphism-as-natural-transformation.md

```diff
@@ -66,6 +80,11 @@ F(t) -> snd
 ```
 
 (note: I believe that this definition requires that every vertex in the graph be both the source of an edge and destination of an edge)
+[Paul: This is related to a concern you expressed on slack: that the construction wouldn't work for DAG's or graphs with isolated vertices.  But everything is fine: the key point is that the functions F(s) and F(t) from the edge set to the vertex set need not be surjective.  I'll illustrate by means of an example.
+
+Consider the graph whose vertext set is V = {1,2,3,4} and whose edge set is E = {e=(1,2), f=(2,3)}.  So this is the path graph on the vertices 1,2,3 (a DAG) together with the isolated vertex 4.  Let us construct the functor X -> *Set* which corresponds to this graph.
+
+As usual we set F(a) = E and F(b) = V.  Define F(s): F(a) -> F(b) by F(s)(e) = 1 and F(s)(f) = 2, and define F(t): F(a) -> F(b) by F(t)(e) = 2 and F(t)(f) = 3. The vertices 3 and 4 do not lie in the range of F(s), and the vertices 1 and 4 do not lie in the range of F(t), but that's ok - these are still well-defined functions.]
```

Comment:
So fundamentally it's ok for the function to produce an image in the target set (which is what you are saying by stating that it doesn't have to be surjective). Great. I had thought that for the structure in category X was two one to one mapping functions (not sure if surjective / bijective is appropriate as it's structure inside the category) they must remain so when mapped. Loosening that constraint makes this a lot easier to reason about.

It kind of makes sense because the object in category X is mapped to a set, not a specific individual value in the set. I had thought that a mapping to a subset would be equivalent to a mapping to a completely different, smaller, set. I guess this isn't the case because of the requirement to be structure preserving it uniquely identifies the target set.